include/envoy/network/address.h

```diff
@@ -125,5 +125,104 @@ class Instance {
 
 typedef std::shared_ptr<const Instance> InstancePtr;
 
+/*
+ * Parse an internet host address (IPv4 or IPv6) and create an Instance from it.
+ * The address must not include a port number.
+ * @param ipAddr string to be parsed as an internet address.
+ * @return pointer to the Instance, or nullptr if unable to parse the address.
+ */
+InstancePtr parseInternetAddress(const std::string& ipAddr);
+
+/*
+ * A range of internet addresses, consisting of an Ip address and a count of leading
+ * bits included in the mask. Other than those leading bits, all of the other bits
+ * of the Ip address are zero.
+ */
+class IpRange {
```

Comment:
Also, since this is not an abstract interface, can we move this into source/common/cidr_range.h|cc ?